Added parameter to callbacks with payload of request

xulture opened the pull request:
Callbacks now get request (UVaRestRequestJSON) as parameter.  C++ signature for callbacks:
void RestRequestComplete(UVaRestRequestJSON* Request);

This essentially makes it easier to track which request returned, in case we are firing off multiple requests while they are still being processed.

![image](https://cloud.githubusercontent.com/assets/3895512/8702595/cc397d9e-2b6c-11e5-8eb8-8adbe5766fe2.png)

ufna commented:
Hi @xulture ,

I've checked the commit files and found that you've deleted FOnRequestFail delegate (that was without parameters), but not implemented one with parameter. I can't check it right now (maybe tomorrow), but it seems a bit strange for me (possible bug?) :)

xulture commented:
Heya Vladimir, thanks for your reply --

Initially I removed that delegate, because OnRequestFail was declared as
FOnRequestComplete;  consequent commit added it back and changed
declaration of OnRequestFail to use it.

Thanks for awesome plugin!!!

Cheers,

Sasha


On Thu, Jul 16, 2015 at 6:40 PM Vladimir Alyamkin <notifications@github.com>
wrote:

> Hi @xulture <https://github.com/xulture> ,
>
> I've checked the commit files and found that you've deleted FOnRequestFail
> delegate (that was without parameters), but not implemented one with
> parameter. I can't check it right now (maybe tomorrow), but it seems a bit
> strange for me (possible bug?) :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ufna/VaRest/pull/31#issuecomment-121848290>.
>

metahusk commented:
SetTimer_Delegate cannot be connected to the events anymore. I haven't had the time to look into it further. Thanks for the addition!

ufna commented:
I've merged the pull request. Thanks @xulture ! :+1:
